constraint/blueprint.go

```diff
@@ -18,11 +18,16 @@ type Blueprint interface {
 	// NbOutputs return the number of output wires this blueprint creates.
 	NbOutputs(inst Instruction) int
 
-	// WireWalker returns a function that walks the wires appearing in the blueprint.
+	// WireWalker returns a function that walks the wires appearing in the blueprint
+	// and optionally the minimum level (minus one) at which the instruction can be solved;
+	// that is; maxLevel = max{level(wire) | wire in blueprint}.
+	// If maxLevel is unknown at compile time, implementation must return 0.
 	// This is used by the level builder to build a dependency graph between instructions.
-	WireWalker(inst Instruction) func(cb func(wire uint32))
+	WireWalker(inst Instruction) (w WireWalker, maxLevel int)
```

Comment:
So, for me it seems correct. One thing which bothers me a bit is that imo this change in lookup blueprint a bit breaks the contract - lookup blueprint does not actually walk all the wires anymore - just the new ones. Maybe we can add an argument to `Blueprint.WireWalker` method if we want to allow caching or not?